### ISSUE TITLE
CheckMeta does not work

### ISSUE THREAD

**Issue by piegamesde** [NixOS/nixpkgs#191124]
I just noticed that meta attributes are actually never checked locally. I noticed that there is an undocumented attribute `config.checkMeta` which defaults to `false`, but even when setting it to true and adding obvious errors to a derivation I cannot get it to fail. Ideally, such an option should be enabled by default IMO.

Arbitrarily pinging @ckiee @Artturin @ncfavier

**Comment by ncfavier** [comment]
Possibly related to https://github.com/NixOS/nixpkgs/pull/6794#issuecomment-1003397767 ?

**Comment by piegamesde** [comment]
That issue is 😱, but I'm not sure this is the one I'm running into since even blatant errors like setting `description = true;` do not fail my evaluation. It really looks like the check is not running at all, but I also can't get it to work somehow.

**Comment by ncfavier** [comment]
Can you provide a minimal reproducer? That much seems to work fine for me:

```nix
with import <nixpkgs> { config.checkMeta = true; };
runCommand "foo" {
  meta.description = true;
} ""
```

```
error: Package ‘foo’ in /home/n/foo.nix:3 has an invalid meta attrset:
       	 - key 'description' has a value 1 of an invalid type bool; expected string, refusing to evaluate.
```

**Comment by ncfavier** [comment]
Can this be closed now?

**Comment by piegamesde** [comment]
I don't think we made any significant progress here, since the original fix got reverted.

**Comment by ConnorBaker** [comment]
Sorry for the bump -- @piegamesde to your knowledge, is `checkMeta` still effectively broken, or has something changed since?

**Comment by piegamesde** [comment]
I don't know of any changes on this

**Comment by mdaniels5757** [comment]
> The derivation will fail with the usual warning, but actually it should fail with "invalid meta" instead.

Why should it fail with an invalid metadata error instead of the insecure/broken error?
